### PR TITLE
[Core][2/N] Helpers for PP

### PR DESCRIPTION
**Pull request by andoorve**
Helpers for https://github.com/vllm-project/vllm/pull/4412

cc: @youkaichao @simon-mo

### CODE REVIEW

File: vllm/distributed/utils.py [hunk 1]
```diff
@@ -134,3 +138,29 @@ def gpu_p2p_access_check(i: int, j: int) -> bool:
         cache = json.load(f)
     _gpu_p2p_access_cache = cache
     return _gpu_p2p_access_cache[f"{i}->{j}"]
+
+
+def get_tensor_model_parallel_src_rank_and_group():
```
Comment (andoorve) [inline vllm/distributed/utils.py:143]: Ideally this function isn't even necesssary and we can change the defaults for `broadcast_tensor_dict` to be TP group broadcast but adding this in to not break the assumption that `broadcast_tensor_dict` uses World group by default.

Comment (andoorve) [reply vllm/distributed/utils.py:143]: We can also return a dict to use as kwargs for the `broadcast_tensor_dict` if this is less ugly:

`broadcast_tensor_dict(**get_tensor_model_parallel_src_rank_and_group())`

**Comment by youkaichao** [comment]
Sorry for the late review. I feel the modification is somewhat intrusive. I plan to refactor the distributed related code in a more OOP-style, and every operation only needs to specify a relative rank, following the RFC https://github.com/vllm-project/vllm/issues/3587 .

e.g. the broadcast tensor dict will be `get_tp().broadcast_tensor_dict(src=0)`, and `get_tp()` will return a tensor parallel coordinator, which knows what does `src=0` mean.

**Comment by andoorve** [comment]
What would you recommend in this case @youkaichao? As this is blocking PP, is it possible to have this merged as it does not affect performance or functionality and then have the refactor you suggest above done at a later time?

**Comment by youkaichao** [comment]
I would recommend adding these helpers after the refactor. There are too many "helper functions" in distributed part, which could been organized in a better way. Let's keep code quality before things get out of control.

My ETA is 1~2 weeks, I originally planned to do the refactor, but was interrupted by the [nccl stuff](https://github.com/vllm-project/vllm/pull/5091) which has a higher priority.

**Comment by andoorve** [comment]
Got it. In that case, to confirm, can we can say we are shelving PP (https://github.com/vllm-project/vllm/pull/4412) until the refactor is done?

**Comment by youkaichao** [comment]
> Got it. In that case, to confirm, can we can say we are shelving PP (#4412) until the refactor is done?

Yes.

**Pull request closed without merging**